clients/src/main/java/org/apache/kafka/common/KafkaFuture.java

```diff
@@ -115,6 +115,15 @@ private void maybeComplete() {
      */
     public abstract <R> KafkaFuture<R> thenApply(BaseFunction<T, R> function);
 
+    /**
+     * Returns a new KafkaFuture that, when this future completes normally, is executed with this
```

Comment:
Do we think that it might be useful to highlight that `thenCompose` aims to chain asynchronous calls in a more natural way as [mentioned](https://issues.apache.org/jira/browse/KAFKA-6900)?

Reply:
I do not know what's best, I think for people who are familiar with Promises/Futures, especially Javas CompletableFuture it's a bit redundant. I think it comes down to philosophy are the javadoc for KafkaFuture meant to explain how KafkaFuture works or to teach people how to use Futures/Promises. If you think I should update the comment I have no problems doing that :)

Reply:
I agree with the point where you mention that it is a bit redundant. I don't think you should update the comment; just wanted to ask the open-ended question to the larger group in case my assumptions around redundancy were incorrect 😄